### ISSUE TITLE
only issue is my ignorance

### ISSUE THREAD

**Issue by GorillaDaddy** [thisisthekap/Xamarin.ExoPlayerSlim#17]
Hello, thanks for posting. I have been trying today to compile a local version, for Android 11 and later, Android 10. Now your project is targeted for Android 12. I'd like to be able to modify the java, for example the code in ExoPlayerSlimImpl. Or add functions to ExoPlayerSlim. I do like the idea of exposing a simple C# interface, and hiding the Android / Exoplayer complexities in Java.

I was able to get it to the point of generating Xamarin.ExoPlayerSlim.dll

But no namespace is exposed when including it in the UsageChecker project, instead of the public nuget Exoplayerslim. 

I thought there might be a .nupkg generated at some point, but there wasn't.

Also counterintuitively, if I modify one of the java files, the DLL isn't regenerated after doing whatever might be necessary to update after that dependency change. Build thinks everything is up-to-date.

Of course the issue here is primarily my ignorance, since this is the first time I've tried to make a java library interface for Xamarin. I was very happy to see you had already done it. It seems I might have to repeat your efforts and start from scratch, and learn how to use Xamarin.binding.helpers and the process described on the microsoft site and whatever other Android Studio stuff that I might need to learn, but that sort of defeats the purpose of your github project.

Any suggestions that might allow me to build and modify and use your github project for Android 11 and Android 10 would be appreciated.

**Comment by thisisthekap** [comment]
Hi @GorillaDaddy,

I think your issue is solved in #19 

@redth updated his Xamarin.Binding.Helpers to work flawlessly again (thank you @redth for your effort).

**Comment by GorillaDaddy** [comment]
> I think your issue is solved in #19
> 

Thank you, but I was never able to get the project fully working/compiling/building-Nuget. Never was able to resolve the automatic building of touched source files, having had to clean/rebuild everything everytime. Twiddling endlessly the previous version, I was eventually able to make some changes and build a package, but when I tried to extend the interface (adding, for example, a getState() to the class) the changes (added public functions) weren't visible to the C# side.

And the latest version (re: #19), I'm unable to get it to build the nuget. I've tried with both Visual Studio 2019 & 2022 (preview & release). Just loading the project cloned from github and attempting a build:
`
 "Your project does not reference "MonoAndroid,Version=v12.0" framework. Add a reference to "MonoAndroid,Version=v12.0" in the "TargetFrameworks" property of your project file and then re-run NuGet restore.	Xamarin.ExoPlayerSlim.UsageChecker"	

I am confident this all works for you. And am disappointed I am unable to reproduce your work. At this point it is just frustrating.

I run Windows 10, tend to install all the latest updates. Android Studio is installed too. I really would like it to work on my dev system, and if you have some time, I would pay you for your time to help me correct whatever problems I am having.
